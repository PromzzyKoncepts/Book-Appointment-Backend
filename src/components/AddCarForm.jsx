/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { useDispatch } from 'react-redux';
// import { useNavigate } from 'react-router-dom';
import { useForm } from 'react-hook-form';
import { toast } from 'react-toastify';
import { createCar } from '../redux/cars/carsSlice';

const AddCarForm = () => {
  // const navigate = useNavigate();
  const dispatch = useDispatch();
  const { register, handleSubmit } = useForm();
  const onSubmit = (car) => {
    console.log(car);
    // JSON.stringify(car);
    const updatedCarData = { ...car, user_id: 1 };
    dispatch(createCar({ updatedCarData, toast }));
  };

  return (
    <form className="add-car-form" onSubmit={handleSubmit(onSubmit)}>
      <input type="text" placeholder="Car Name" {...register('name', { required: true, mainLength: 3 })} />
      <input type="text" placeholder="Car Model" {...register('model', { required: true, mainLength: 3 })} />
      <input type="number" placeholder="Price" {...register('price', { required: true })} />
      <input type="text" placeholder="Image url" {...register('image_url')} />
      <textarea type="text" placeholder="Description" {...register('description', { required: true })} />
      <input type="submit" />
    </form>
  );
};

export default AddCarForm;
