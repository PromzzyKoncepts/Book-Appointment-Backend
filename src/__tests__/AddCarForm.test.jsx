import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import store from '../redux/configureStore';
import '@testing-library/jest-dom/';
import AddCarForm from '../components/AddCarForm';

describe('AddCarForm component', () => {
  test('Should render the proper AddCarForm component', () => {
    const addCarFormcomponent = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <AddCarForm />
          </Router>
        </Provider>
        ,
      </React.StrictMode>,
    );
    expect(addCarFormcomponent).toMatchSnapshot();
  });
});
