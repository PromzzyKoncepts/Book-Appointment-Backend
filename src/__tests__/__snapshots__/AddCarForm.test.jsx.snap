// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCarForm component Should render the proper AddCarForm component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gueYoa hjwgPs"
      >
        <form
          class="sc-beqWaB eTVxgX"
        >
          <h1
            class="sc-gLDzan hoMnOv"
          >
            Add a car
          </h1>
          <input
            class="sc-dmqHEX jEDZgU form-control"
            name="name"
            placeholder="Enter name"
            required=""
            type="text"
            value=""
          />
          <input
            class="sc-dmqHEX jEDZgU form-control"
            name="model"
            placeholder="Enter car model"
            required=""
            type="text"
            value=""
          />
          <input
            class="sc-dmqHEX jEDZgU form-control"
            name="price"
            placeholder="Enter Price"
            required=""
            type="number"
            value="0"
          />
          <input
            class="sc-dmqHEX jEDZgU form-control"
            name="image_url"
            placeholder="Enter Image URL"
            required=""
            type="text"
            value=""
          />
          <textarea
            class="sc-eDDNvR gOqxBO form-control"
            name="description"
            placeholder="Enter description"
            required=""
            type="text"
          />
          <button
            class="sc-jTrPJq GpDmH"
            type="submit"
          >
            Add Car
          </button>
        </form>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gueYoa hjwgPs"
    >
      <form
        class="sc-beqWaB eTVxgX"
      >
        <h1
          class="sc-gLDzan hoMnOv"
        >
          Add a car
        </h1>
        <input
          class="sc-dmqHEX jEDZgU form-control"
          name="name"
          placeholder="Enter name"
          required=""
          type="text"
          value=""
        />
        <input
          class="sc-dmqHEX jEDZgU form-control"
          name="model"
          placeholder="Enter car model"
          required=""
          type="text"
          value=""
        />
        <input
          class="sc-dmqHEX jEDZgU form-control"
          name="price"
          placeholder="Enter Price"
          required=""
          type="number"
          value="0"
        />
        <input
          class="sc-dmqHEX jEDZgU form-control"
          name="image_url"
          placeholder="Enter Image URL"
          required=""
          type="text"
          value=""
        />
        <textarea
          class="sc-eDDNvR gOqxBO form-control"
          name="description"
          placeholder="Enter description"
          required=""
          type="text"
        />
        <button
          class="sc-jTrPJq GpDmH"
          type="submit"
        >
          Add Car
        </button>
      </form>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
